### ISSUE TITLE
Support for passing compiler options

### ISSUE THREAD

**Issue by RajivKurian** [FredTingaud/quick-bench-front-end#3]
Thanks for the great project. It would be great if we could pass compiler options for our benchmark  like gcc.godbolt.org. Right now I want to run my benchmark with -mavx2 and I can't find a way to do that.

**Comment by FredTingaud** [comment]
Hi,
I thought about it, but the problem with passing compiler options like that is that it would require some serious sanitizing, to avoid injections.
I am not very clear on how I could do it. I will have a look at godbolt's solution and see if I can use it.

**Comment by RajivKurian** [comment]
Yeah, looking at godbolt's solution would be my suggestion too.

An alternative might be to have a few selectable options like -march=native
etc, like you do for optimization level.

Thanks!

**Comment by didito** [comment]
Yes, being able to select architecture would help in benchmarking SIMD intrinsic code!

**Comment by kgariful** [comment]
Hi, -march option would be extremely useful. I make a benchmarking with SIMD instructions and autovectorization

**Comment by alexander-titov** [comment]
Hi,
It is a highly desirable feature!

**Comment by FredTingaud** [comment]
Hi,
Quick Bench runs on AWS, that doesn't guarantee any architecture for the kind of machines the project can afford.
Thus it would not possible to consistently run tests compiled with a given -march flags.

**Comment by Yankes** [comment]
There is workaround, if you use `__attribute__ ((target("fma"), optimize("-ffast-math")))` you can force GCC to change target processor per function and use SSE e.g.: http://quick-bench.com/SBnGN_2uHuwiFi5nIzekFVKK3ic
One important caveat is that test function need same attribute otherwise GCC will not inline it.

**Comment by didito** [comment]
Awesome, thank you @Yankes - that seems to work.

here are links to the documentation of the target attribute:
https://gcc.gnu.org/onlinedocs/gcc-8.2.0/gcc/x86-Function-Attributes.html#x86-Function-Attributes
https://clang.llvm.org/docs/AttributeReference.html#target-gnu-target

and here are the most relevant values:
- sse3
- ssse3
- sse4
- sse4.1
- sse4.2
- sse4a
- fma4
- avx

**Comment by FredTingaud** [comment]
Thanks for your comments!
Right now, I am super busy and can't really experiment with this, but I'll try to check that next time I work on an update.

**Comment by MattPD** [comment]
Support for `-Ofast`  (and perhaps `-Og`) among the `optim` options would be valuable as well.

**Comment by yvef** [comment]
As far as I see the topic is pretty old, but still relevant.
Is there any chance of changes?
Thanks.

**Comment by stephenberry** [comment]
You could always check a preset list of compiler flags.
It would be very helpful to be able to benchmark intrinsics (e.g. `-mbmi2`), otherwise we can't really test peak performance.

**Comment by Guiorgy** [comment]
@FredTingaud As a stop-gap, you could allow passing compiler options when run locally for those that need. Would that be difficult?

**Comment by bytesiz3d** [comment]
Bumping as I have run into the exact same problem. Need to benchmark AVX2 intrinsics.
